app/src/main/java/nerd/tuxmobil/fahrplan/congress/alarms/AlarmReceiver.kt

```diff
@@ -0,0 +1,211 @@
+package nerd.tuxmobil.fahrplan.congress.alarms
+
+import nerd.tuxmobil.fahrplan.congress.utils.PendingIntentCompat.FLAG_IMMUTABLE
+import nerd.tuxmobil.fahrplan.congress.contract.BundleKeys
+import nerd.tuxmobil.fahrplan.congress.schedule.MainActivity.Companion.createLaunchIntent
+import nerd.tuxmobil.fahrplan.congress.exceptions.BuilderException
+import nerd.tuxmobil.fahrplan.congress.autoupdate.UpdateService.Companion.start
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository
+import nerd.tuxmobil.fahrplan.congress.notifications.NotificationHelper
+
+import info.metadude.android.eventfahrplan.commons.logging.Logging
+
+import android.net.Uri
+import android.app.PendingIntent
+import android.content.Intent
+import android.content.Context
+import android.content.BroadcastReceiver
+import androidx.annotation.VisibleForTesting
+
+class AlarmReceiver : BroadcastReceiver() {
+
+    companion object {
+        const val ALARM_UPDATE = "nerd.tuxmobil.fahrplan.congress.ALARM_UPDATE"
+        const val ALARM_SESSION = "nerd.tuxmobil.fahrplan.congress.ALARM_SESSION"
+        const val ALARM_DELETE = "de.machtnix.fahrplan.ALARM"
+        const val ALARM_DISMISSED = "nerd.tuxmobil.fahrplan.congress.ALARM_DISMISSED"
+        private const val LOG_TAG = "AlarmReceiver"
+        private const val BUNDLE_KEY_NOTIFICATION_ID = "BUNDLE_KEY_NOTIFICATION_ID"
+        private const val INVALID_NOTIFICATION_ID = -1
+        private const val DEFAULT_REQUEST_CODE = 0
+
+        /**
+         * Returns a unique [Intent] to delete the data associated with the
+         * given session alarm `notificationId`.
+         * The `Intent` is composed with a fake data URI to comply with the uniqueness rules
+         * defined by [Intent.filterEquals].
```

Comment:
KDoc: Please use square brackets for all variables or classes being referenced instead of backticks.